sklearn/tree/_tree.pyx

```diff
@@ -1510,11 +1510,11 @@ cdef _cost_complexity_prune(unsigned char[:] leaves_in_subtree, # OUT
 
             # bubble up values to ancestor nodes
             current_r = r_node[leaf_idx]
-            while leaf_idx != 0:
-                parent_idx = parent[leaf_idx]
-                r_branch[parent_idx] += current_r
-                n_leaves[parent_idx] += 1
-                leaf_idx = parent_idx
+            node_idx = parent[leaf_idx]
+            while node_idx != _TREE_UNDEFINED:
+                r_branch[node_idx] += current_r
+                n_leaves[node_idx] += 1
+                node_idx = parent[node_idx]
```

Comment:
This changes the order a little bit, but the computation is the same. Originally the logic was:

> While not root node (`leaf_idx != 0`), then update the parent.

Now the logic is:

> While the node is defined, then update yourself.